### PR TITLE
Adding operator-workflows

### PR DESCRIPTION
**Pull request by gtrkiller**
This PR replaces the old workflows with those standardized on https://github.com/canonical/operator-workflows

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -1,28 +1,26 @@
...
-    steps:
-      - uses: actions/checkout@v3
-      - name: Release charm to channel
-        uses: canonical/charming-actions/release-charm@1.0.3
-        with:
-          credentials: ${{ secrets.CHARMHUB_TOKEN }}
-          github-token: ${{ secrets.GITHUB_TOKEN }}
-          destination-channel: ${{ github.event.inputs.destination-channel }}
-          origin-channel: ${{ github.event.inputs.origin-channel }}
+    uses: canonical/operator-workflows/.github/workflows/promete_charm.yaml@main
```
Comment (gregory-schiano) [inline .github/workflows/release.yaml:22]: Is there a typo here ?
Should be `promote_charm.yaml@main`

File: microk8s-installation.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+sudo microk8s status --wait-ready
+sudo microk8s enable dns hostpath-storage registry
+sudo usermod -a -G microk8s $USER
+sudo chown -f -R $USER ~/.kube
+sudo microk8s kubectl -n kube-system rollout status -w deployment/hostpath-provisioner
+sudo microk8s kubectl -n kube-system rollout status -w deployment/coredns
+sudo microk8s kubectl -n container-registry rollout status -w deployment/registry
+sg microk8s -c "juju bootstrap microk8s k8s-ctrl"
+juju add-model testing
+docker load --input gunicorn/gunicorn.tar
```
Comment (gregory-schiano) [inline microk8s-installation.sh:12]: Where does the tar come from ?

Comment (gtrkiller) [reply microk8s-installation.sh:12]: From the build image workflow, it builds the image in that location.

File: .github/workflows/release.yaml (hunk 1 above)
Comment (gtrkiller) [reply .github/workflows/release.yaml:22]: Ty! correcting it right away

File: microk8s-installation.sh (hunk 2 above)
Comment (gtrkiller) [reply microk8s-installation.sh:12]: the thing is that, as I'm using lxd as a provider, I have to load the image manually

Comment (gregory-schiano) [reply microk8s-installation.sh:12]: Wanted to make sure I got it properly, make sense

**Review by merkata — APPROVED** [review]
Looks good, great job!

File: microk8s-installation.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/bash
+sudo snap install microk8s --classic
+sudo microk8s status --wait-ready
+sudo microk8s enable dns hostpath-storage registry
+sudo usermod -a -G microk8s $USER
+sudo chown -f -R $USER ~/.kube
+sudo microk8s kubectl -n kube-system rollout status -w deployment/hostpath-provisioner
+sudo microk8s kubectl -n kube-system rollout status -w deployment/coredns
+sudo microk8s kubectl -n container-registry rollout status -w deployment/registry
+sg microk8s -c "juju bootstrap microk8s k8s-ctrl"
```
Comment (amandahla) [inline microk8s-installation.sh:10]: Wouldnt be necessary to run juju switch k8s-ctrl  before add-model??
I think integration_test.yaml had this command.

Comment (gtrkiller) [reply microk8s-installation.sh:10]: It is not. When you run juju bootstrap, you will be on that controller by default after the bootstrap ends. You would need to execute a switch if you need to bootstrap a controller and then switch to another one.

Comment (merkata) [reply microk8s-installation.sh:10]: You might want to put `$USER` in quotes like `"$USER"` to keep shellcheck happy, although not required.

**Review by merkata — APPROVED** [review]

**Review by amandahla — APPROVED** [review]

**Pull request closed without merging**